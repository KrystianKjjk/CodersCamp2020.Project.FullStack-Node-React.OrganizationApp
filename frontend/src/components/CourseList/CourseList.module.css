.listContainer{
    padding-top: 2%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    background-color: #1C1C1C;
    border: 1px solid #666666;
}

.manageCourseBar{
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color:#1C1C1C;
    border: 1px solid #666666;
    border-bottom: none;
    padding: 0 3%;
}
.manageCourseBar button{
    background-color: #2196F3;
    height: 50%;
    width: 8%;
}
.manageCourseBar button:hover{
    background-color:#67b5ff;
}