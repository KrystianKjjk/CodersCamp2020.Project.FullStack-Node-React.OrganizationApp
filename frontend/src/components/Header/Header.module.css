@import url('https://fonts.googleapis.com/css2?family=Sarpanch:wght@600&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@500&family=Sarpanch:wght@600&display=swap');
.header {
  width: 100%;
  padding: 10px 0;
  background-color: #292929;
  color: #ffffff;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid #666666;
}

.logo {
  margin-left: 2rem;
  font-family: 'Sarpanch', sans-serif;
  font-size: 2rem;
}

.logo span {
  color: #ffc53d;
}

.logout {
  margin-right: 2rem;
  margin-left: 3rem;
  font-family: 'Roboto', sans-serif;
  display: flex;
  align-items: center;
  cursor: pointer;
}

.logout span {
  padding-left: 10px;
}

.activeCourseBox {
  display: flex;
  font-family: 'Roboto', sans-serif;
}

.activeCourse {
  color: #ffc53d;
  margin-left: 1rem;
}
