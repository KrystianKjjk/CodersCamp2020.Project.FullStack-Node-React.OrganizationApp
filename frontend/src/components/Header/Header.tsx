import React from "react";
import styles from "./Header.module.css";
import PowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import { removeUserFromLocalStorage } from "../../app/utils";
import {useHistory} from "react-router-dom";
import { useAppSelector } from "../../app/hooks";

interface HeaderProps {
  onLogout?: Function;
}

const Header = (props: HeaderProps) => {
  // const Header: React.FC = () => {
  const { activeCourse } = useAppSelector((state) => state.courseList);
  const handleLogOut = () => {
    removeUserFromLocalStorage();
  };

  const history = useHistory();
  const routeChange = () => {
    let path = `/`;
    history.push(path);
  };

    const takeHome = () => {
        let path = `/home`;
        history.push(path);
    };

  return (
    <div className={styles.header}>
      <div className={styles.logo} onClick={takeHome} style={{cursor: "pointer"}}>
            <span>.</span>Coders<span>Camp</span>
      </div>
      <div className={styles.activeCourseBox}>
        <p className={styles.activeCourse}>{activeCourse?.name}</p>
        <div
          className={styles.logout}
          onClick={() => {
            handleLogOut();
            routeChange();
            if (props.onLogout) props.onLogout();
          }}
        >
          <PowerSettingsNewIcon
            style={{ color: "rgba(255, 255, 255, 0.6)" }}
          ></PowerSettingsNewIcon>
          <span> Log out</span>
        </div>
      </div>
    </div>
  );
};

export default Header;
