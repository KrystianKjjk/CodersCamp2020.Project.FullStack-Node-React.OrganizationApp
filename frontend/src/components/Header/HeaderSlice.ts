import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppThunk, RootState } from '../../app/store';

interface HeaderState {
  
}

const initialState: HeaderState = {
  
};

export const headerSlice = createSlice({
  name: 'header',
  initialState,
  reducers: {

  },
});

export const { } = headerSlice.actions;

// if you want, add selectors here, change the one below, remember to register reducer in store.ts
// export const selectHeader = (state: RootState) => state.header.value;

export default headerSlice.reducer;
