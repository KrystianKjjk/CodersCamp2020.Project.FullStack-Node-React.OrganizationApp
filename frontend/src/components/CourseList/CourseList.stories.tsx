import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import CourseList , { CourseListProps } from './CourseList';
import { store } from '../../app/store';
import { Provider } from 'react-redux';

export default {
  title: 'CourseList component',
  component: CourseList,
} as Meta;

const Template: Story<CourseListProps> = (args) => (
  <Provider store={store}>
    <CourseList {...args} />
  </Provider>
);

export const SampleCourseList = Template.bind({});
SampleCourseList.args = {
  name: 'CourseList',
};