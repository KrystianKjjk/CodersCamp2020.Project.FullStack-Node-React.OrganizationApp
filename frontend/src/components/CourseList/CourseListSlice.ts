import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { AppThunk, RootState } from "../../app/store";
import { CourseCreateObject } from "../Course/CourseDetailsSlice";
import {
  CourseListElementDto,
  deleteCourse,
  fetchCourses,
} from "../Course/CourseClient";
import CourseList from "./CourseList";
import {getActiveCourse, setActiveCourse} from "../../app/ActiveCourse";
import CoursesService from "../../api/courses.service";
// import CourseCreateObject from '../CourseCreate';

export interface CourseListElementModel extends CourseCreateObject {
  _id: string;
}

interface CourseListState {
  courseList: CourseListElementModel[];
}

const initialState: CourseListState = {
  courseList: [],
};

export const fetchCoursesAsync = (): AppThunk => async (dispatch) => {
  const courses=await fetchCoursesAndSort();
  dispatch(setCourses(courses));
};

export const fetchCoursesAndSort=async ()=>{
  const courseService = new CoursesService();
  const response = await courseService.fetchCourses();
  // const response = await fetchCourses();
  const coursesDto: CourseListElementDto[] = response.data;
  const courses: CourseListElementModel[] = coursesDto.map((courseDto) => {
    const courseListElement: CourseListElementModel = {
      _id: courseDto._id,
      name: courseDto.name,
      description: courseDto.description,
      endDate: new Date(Date.parse(courseDto.endDate)),
      startDate: new Date(Date.parse(courseDto.startDate)),
    };
    return courseListElement;
  });
  courses.sort(function (courseListElement1, courseListElement2) {
    var courseListElementStartDate1 = courseListElement1.startDate.getTime();
    var courseListElementStartDate2 = courseListElement2.startDate.getTime();
    return courseListElementStartDate2 - courseListElementStartDate1;
  });
  return courses;
}

export const deleteCourseAsync = (courseId: string): AppThunk => async (
  dispatch
) => {
  deleteCourse(courseId).then(() => dispatch(removeCourse(courseId)));
};

export const courseListSlice = createSlice({
  name: "courseList",
  initialState,
  reducers: {
    setCourses: (state, action: PayloadAction<CourseListElementModel[]>) => {
      state.courseList = action.payload;
    },
    removeCourse: (state, action: PayloadAction<string>) => {
      const idToDelete = action.payload;
      state.courseList = state.courseList.filter(
        (courseListElement) => courseListElement._id !== idToDelete
      );
    },
  },
});

export const { setCourses, removeCourse } = courseListSlice.actions;

// if you want, add selectors here, change the one below, remember to register reducer in store.ts
// export const selectCourseList = (state: RootState) => state.courseList.value;

export default courseListSlice.reducer;

