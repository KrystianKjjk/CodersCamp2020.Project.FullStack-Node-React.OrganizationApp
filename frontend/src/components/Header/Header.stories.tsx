import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import Header from './Header'

export default {
  title: 'Header component',
  component: Header,
} as Meta

const Template: Story = (args) => <Header {...args} />

export const BasicHeader = Template.bind({})
