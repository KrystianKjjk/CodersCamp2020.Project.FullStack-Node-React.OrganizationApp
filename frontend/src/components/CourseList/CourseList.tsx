import React, { useEffect } from "react";
import styles from "./CourseList.module.css";
import { fetchCoursesAsync } from "./CourseListSlice";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import CourseListElement from "../CourseList/CourseListElement";
import PageHeader from "../PageHeader";
import { Button } from "@material-ui/core";
import { useHistory } from "react-router-dom";
import UButton from "../UButton";

export interface CourseListProps {}

const CourseList: React.FC<CourseListProps> = (props) => {
  const dispatch = useAppDispatch();
  const history = useHistory();
  const { courseList, activeCourse } = useAppSelector(
    (state) => state.courseList
  );

  useEffect(() => {
    dispatch(fetchCoursesAsync());
  }, [dispatch]);

  const listElements = courseList.map((courseListElement) => (
    <CourseListElement
      key={courseListElement._id}
      course={courseListElement}
      isActive={courseListElement._id === activeCourse?._id}
    />
  ));

  const handleAddButtonClick = (event: any) => {
    history.push("coursecreate/");
  };

  return (
    <div>
      <PageHeader name="COURSES"></PageHeader>
      <div>
        <div className={styles.manageCourseBar}>
          <h3>Manage courses</h3>
          <UButton
            text="ADD"
            color="primary"
            onClick={handleAddButtonClick}
          ></UButton>
        </div>
        <div className={styles.listContainer}>{listElements}</div>
      </div>
    </div>
  );
};

export default CourseList;
