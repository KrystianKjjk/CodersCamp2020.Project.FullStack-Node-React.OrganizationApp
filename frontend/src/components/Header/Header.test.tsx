import React from "react";
import Header from "./index";
import { render} from "@testing-library/react";
import * as Hooks from "../../app/hooks";
import { CourseListElementModel } from "../../pages/Admin/CourseList/CourseListSlice";

const courseListElement: CourseListElementModel = {
  _id: "dummyId",
  name: "dummyName",
  description: "dummyDescription",
  endDate: new Date(),
  startDate: new Date(),
};

describe("Header", () => {
  it("should render header", () => {
    jest.spyOn(Hooks,"useAppSelector").mockReturnValue({activeCourse:courseListElement})
   const {container}= render(<Header />);
    expect(container).toMatchSnapshot()
  });
});
