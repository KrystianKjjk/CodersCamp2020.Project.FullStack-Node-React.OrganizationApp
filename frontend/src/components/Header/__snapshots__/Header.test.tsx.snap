// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="logo"
    >
      <span>
        .
      </span>
      Coders
      <span>
        Camp
      </span>
    </div>
    <div
      class="activeCourseBox"
    >
      <p
        class="activeCourse"
      >
        dummyName
      </p>
      <div
        class="logout"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="color: rgba(255, 255, 255, 0.6);"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
          />
        </svg>
        <span>
           Log out
        </span>
      </div>
    </div>
  </div>
</div>
`;
